Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -9414,6 +9414,28 @@ void WebPage::createTextFragmentDirectiveFromSelection(CompletionHandler<void(UR
     completionHandler(WTFMove(url));
 }
 
+void WebPage::getTextFragmentRanges(CompletionHandler<void(const Vector<EditingRange>&&)>&& completionHandler)
+{
+    RefPtr focusedOrMainFrame = protectedCorePage()->checkedFocusController()->focusedOrMainFrame();
+    if (!focusedOrMainFrame)
+        return;
```

Comment:
You should still call the completion handler before early returning:

```suggestion
    if (!focusedOrMainFrame) {
        completionHandler({ });
        return;
    }
```